common/djangoapps/third_party_auth/api/views.py

```diff
@@ -89,3 +98,122 @@ def get(self, request, username):
+        linked to the given backend.
+
+        The list can be filtered by edx username or third party ids. The filter is limited by the max length of URL.
+        It is suggested to query no more than 50 usernames or remote_ids in each request to stay within above
+        limitation
+
+        Each page in the list can contain up to 50 mappings.
+
+    **Example Requests**
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users?usernames={username1},{username2}
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users?remote_ids={remote_id1},{remote_id2}
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users?usernames={username1}&remote_ids={remote_id1}
+
+    **URL Parameters**
+
+        * provider_id: The unique identifier of third_party_auth provider (e.g. "saml-ubc", "oa2-google", etc.
+          This is not the same thing as the backend_name.). (Optional/future: We may also want to allow
+          this to be an 'external domain' like 'ssl:MIT' so that this API can also search the legacy
+          ExternalAuthMap table used by Standford/MIT)
+
+    **Query Parameters**
+
+        * remote_ids: Optional. List of comma separated remote (third party) user IDs to filter the result set.
+          e.g. ?remote_ids=8721384623
+
+        * usernames: Optional. List of comma separated edX usernames to filter the result set.
+          e.g. ?usernames=bob123,jane456
+
+        * page, page_size: Optional. Used for paging the result set, especially when getting
+          an unfiltered list.
+
+    **Response Values**
+
+        If the request for information about the user is successful, an HTTP 200 "OK" response
+        is returned.
+
+        The HTTP 200 response has the following values:
+
+        * count: The number of mappings for the backend.
+
+        * next: The URI to the next page of the mappings.
+
+        * previous: The URI to the previous page of the mappings.
+
+        * num_pages: The number of pages listing the mappings.
+
+        * results:  A list of mappings returned. Each collection in the list
+          contains these fields.
+
+            * username: The edx username
+
+            * remote_id: The Id from third party auth provider
+    """
+    authentication_classes = (
+        OAuth2Authentication,
+    )
+
+    paginate_by = 50
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.UserMappingSerializer
+    provider = None
+
+    def get_queryset(self):
+        self.provider = Registry.get(self.kwargs.get('provider_id'))
+        if not self.provider:
+            raise Http404
+
+        # permission checking. We allow both API_KEY access and OAuth2 client credential access
+        if not (
+                ApiKeyHeaderPermission().has_permission(self.request, self) or
```

Comment:
I added superuser logic. However, I'm not sure if it is useful since it's intended usage is for server-to-server. And the logic for superuser and API_KEY in `UserView.get` is `or`. It allows the user itself, superuser or request with API_KEY to query.